lib/views/post/widgets/vote_button.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:flutter/material.dart';
+import 'package:pr0gramm/constants/vote_constants.dart';
+import 'package:pr0gramm/entities/enums/vote.dart';
+import 'package:pr0gramm/entities/enums/vote_button_type.dart';
+import 'package:pr0gramm/services/vote_animation_service.dart';
+
+class VoteButton extends StatefulWidget {
```

Comment:
Make this an abstract class and implement individual components for UpVoteButton, DownVoteButton and FavoriteVoteButtonThis get's rid of the huge onStateChange method (as each part is implemented in the inidividual buttons), as well as the VoteButtonType enum.

Reply:
type still needed

Reply:
or abuse Vote to define a button further because at some point the button needs to be mapped to the state in the animation service

Reply:
and the service needs the accosiation between which registered button is what Vote

Reply:
onStateChange still needed because this is the entry point for a button how dont know anythig about the other buttons to get the info what to display currently.
so the onStateChange  method will stil be needed but only for the specific logic for the Specific Implementation of VoteButton
And i think mixin for the state class of "Implementation of VoteButton" is should be fine: containing all the redundant functionality

Reply:
dont need an abstract class

Reply:
I'm counting at least 40 lines and dozens if statements / branches that handle the specific up/down/favorite logic. This is definitely the point to split it into individual components. If you are solving it with composition or inheritance is up to you.

Reply:
yes splitting makes sence

Reply:
is resolved